File: subutil/pkg/src/main/java/com/blankj/subutil/pkg/feature/SubUtilActivity.java
package com.blankj.subutil.pkg.feature;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.view.View;

import com.blankj.lib.base.BaseBackActivity;
import com.blankj.subutil.pkg.R;
import com.blankj.subutil.pkg.feature.brightness.BrightnessActivity;
import com.blankj.subutil.pkg.feature.flashlight.FlashlightActivity;
import com.blankj.subutil.pkg.feature.location.LocationActivity;
import com.blankj.subutil.pkg.feature.pinyin.PinyinActivity;
import com.blankj.utilcode.util.BusUtils;

/**
 * <pre>
 *     author: Blankj
 *     blog  : http://blankj.com
 *     time  : 2016/09/29
 *     desc  : MainActivity
 * </pre>
 */
public class SubUtilActivity extends BaseBackActivity {

    @BusUtils.Subscribe(name = "SubUtilActivity#start")
    public static void start(Context context) {
        Intent starter = new Intent(context, SubUtilActivity.class);
        context.startActivity(starter);
    }

    @Override
    public void initData(@Nullable Bundle bundle) {

    }

    @Override
    public int bindLayout() {
        return R.layout.activity_util_sub;
    }

    @Override
    public void initView(Bundle savedInstanceState, View contentView) {
        setTitle(R.string.sub_util);
    }

    @Override
    public void doBusiness() {

    }

    @Override
    public void onWidgetClick(View view) {

    }

    public void brightnessClick(View view) {
        BrightnessActivity.start(this);
    }

    public void flashlightClick(View view) {
        FlashlightActivity.start(this);
    }

    public void locationClick(View view) {
        LocationActivity.start(this);
    }

    public void pinyinClick(View view) {
        PinyinActivity.start(this);
    }
}

File: subutil/pkg/src/main/java/com/blankj/subutil/pkg/feature/SubUtilActivity.kt
package com.blankj.subutil.pkg.feature

import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.view.View
import com.blankj.lib.common.CommonTitleActivity
import com.blankj.subutil.pkg.R
import com.blankj.subutil.pkg.feature.appStore.AppStoreActivity
import com.blankj.subutil.pkg.feature.countryCode.CountryCodeActivity
import com.blankj.subutil.pkg.feature.location.LocationActivity
import com.blankj.subutil.pkg.feature.pinyin.PinyinActivity
import com.blankj.subutil.util.AppStoreUtils
import com.blankj.utilcode.util.BusUtils
import kotlinx.android.synthetic.main.activity_util_sub.*
import kotlinx.android.synthetic.main.activity_util_sub.view.*

/**
 * ```
 * author: Blankj
 * blog  : http://blankj.com
 * time  : 2016/09/29
 * desc  : MainActivity
 * ```
 */
class SubUtilActivity : CommonTitleActivity() {

    companion object {
        @BusUtils.Subscribe(name = "SubUtilActivity#start")
        fun start(context: Context) {
            val starter = Intent(context, SubUtilActivity::class.java)
            context.startActivity(starter)
        }
    }

    override fun bindTitle(): CharSequence {
        return getString(R.string.sub_util)
    }

    override fun initData(bundle: Bundle?) {}

    override fun bindLayout(): Int {
        return R.layout.activity_util_sub
    }

    override fun initView(savedInstanceState: Bundle?, contentView: View?) {
        applyDebouncingClickListener(
                subUtilAppStoreBtn,
                subUtilCountryCodeBtn,
                subUtilLocationBtn,
                subUtilPinyinBtn
        )
    }

    override fun doBusiness() {}

    override fun onDebouncingClick(view: View) {
        when(view.id) {
            R.id.subUtilAppStoreBtn -> AppStoreActivity.start(this)
            R.id.subUtilCountryCodeBtn -> CountryCodeActivity.start(this)
            R.id.subUtilLocationBtn -> LocationActivity.start(this)
            R.id.subUtilPinyinBtn -> PinyinActivity.start(this)
        }
    }
}
